feature/google/src/main/java/com/revenuecat/purchases/google/findBestOffer.kt

```diff
@@ -0,0 +1,108 @@
+@file:Suppress("MaximumLineLength", "MaxLineLength")
+package com.revenuecat.purchases.google
+
+import com.revenuecat.purchases.models.GoogleSubscriptionOption
+import com.revenuecat.purchases.models.PricingPhase
+import com.revenuecat.purchases.models.RecurrenceMode
+
+fun List<GoogleSubscriptionOption>.findBestOffer(): GoogleSubscriptionOption? {
+    val basePlan = this.firstOrNull { it.isBasePlan } ?: return null
+
+    val validOffers = this
+        .filter { !it.isBasePlan }
+        .filter { !it.tags.contains("rc-ignore-best-offer") }
+
+    return findLongestFreeTrial(validOffers) ?: findBestSavingsOffer(basePlan, validOffers) ?: basePlan
+}
+
+private fun findLongestFreeTrial(offers: List<GoogleSubscriptionOption>): GoogleSubscriptionOption? {
+    return offers.mapNotNull { offer ->
+        offer.pricingPhases.filter { pricingPhase ->
+            pricingPhase.priceAmountMicros == 0L
+        }.map {
+            Pair(offer, parseBillPeriodToDays(it.billingPeriod))
+        }.maxByOrNull { it.second }
+    }.maxByOrNull { it.second }?.first
+}
+
+@Suppress("ReturnCount")
+private fun findBestSavingsOffer(
+    basePlan: GoogleSubscriptionOption,
+    offers: List<GoogleSubscriptionOption>
+): GoogleSubscriptionOption? {
+    val basePlanPricingPhase = basePlan.pricingPhases.firstOrNull() ?: return null
+
+    val longestOfferTotalDays = offers
+        .mapNotNull { subscriptionOption ->
+            subscriptionOption.pricingPhases.sumOf { it.totalNumberOfDays() ?: 0 }
+        }.maxByOrNull { it } ?: 0
+
+    if (longestOfferTotalDays < 0) {
+        return null
+    }
+
+    val periodDays = parseBillPeriodToDays(basePlanPricingPhase.billingPeriod)
+    val basePlanCostPerDay = basePlanPricingPhase.priceAmountMicros / periodDays
+
+    return offers.map { offer ->
+        // Maps each phase for total cost up to longest offer duration
+        val pricePerDayPerPhase = offer.pricingPhases
+            .filter { it.recurrenceMode == RecurrenceMode.FINITE_RECURRING }
```

Comment:
ooo this is interesting, so it's really only mapping non-base-plan phases here. maybe we can update comments/naming to reflect

Reply:
This code is no longer a thing because of reworked logic 🤷‍♂️